scripts/ir2cpp_runner_before.cc

```diff
@@ -0,0 +1,127 @@
+#include <iostream>
+#include <map>
+#include <cstdarg>
+
+// TODO: Not a relative path, of course.
+#include "../current/blocks/json/json.h"
+
+using namespace current::json;
+
+struct value_t final {
+  JSONValue opa_value;  // TODO: OPA's `undefined` is not the same as `null`, of course.
+
+  value_t() : opa_value(JSONNull()) {}
+  value_t(JSONValue opa_value) : opa_value(std::move(opa_value)) {}
+
+  void ResetToUndefined() {
+    opa_value = JSONNull();
+  }
+
+  bool IsUndefined() const {
+    return Exists<JSONNull>(opa_value);
+  }
+
+  bool IsString(char const* desired) const {
```

Comment:
Hmm, the name hints `IsString` is same as `IsUndefined`, but it's actually `IsStringEqualTo`. I understand it's OPA's internal making, but are we forced to use it everywhere? TODO(dkorolev): Speak with @anderseknert & team about this.

Reply:
Hey there! :) Where does `IsString` come from? I don't see it as part as part of the IR specification, and except for the OPA builtin `is_string` (which checks only for type, not value) I can't find any references to this in the OPA codebase either.

Reply:
Ah, right, thanks! Taken care of in the last (two) commits of https://github.com/C5T/jsopa/pull/5 :-)